modules/audigentRtdProvider.js

```diff
@@ -26,116 +24,112 @@ const storage = getStorageManager();
 
 /** @type {string} */
 const MODULE_NAME = 'realTimeData';
-
-/** @type {ModuleParams} */
-let _moduleParams = {};
+const SUBMODULE_NAME = 'audigent';
 
 /**
  * XMLHttpRequest to get data form audigent server
  * @param {string} url server url with query params
  */
-
 export function setData(data) {
   storage.setDataInLocalStorage('__adgntseg', JSON.stringify(data));
 }
 
-function getSegments(adUnits, onDone) {
-  try {
-    let jsonData = storage.getDataFromLocalStorage('__adgntseg');
-    if (jsonData) {
-      let data = JSON.parse(jsonData);
-      if (data.audigent_segments) {
-        let dataToReturn = adUnits.reduce((rp, cau) => {
-          const adUnitCode = cau && cau.code;
-          if (!adUnitCode) { return rp }
-          rp[adUnitCode] = data;
-          return rp;
-        }, {});
-
-        onDone(dataToReturn);
-        return;
-      }
+function addSegmentData(adUnits, data) {
+  adUnits.forEach(adUnit => {
+    if (adUnit.hasOwnProperty('bids')) {
+      adUnit.bids.forEach(bid => {
+        bid.audigent_segments = data;
+      })
     }
-    getSegmentsAsync(adUnits, onDone);
-  } catch (e) {
-    getSegmentsAsync(adUnits, onDone);
-  }
+  })
+
+  return adUnits;
 }
 
-function getSegmentsAsync(adUnits, onDone) {
+function getSegments(reqBidsConfigObj, onDone, config, userConsent) {
+  const adUnits = reqBidsConfigObj.adUnits || getGlobal().adUnits;
+
+  let jsonData = storage.getDataFromLocalStorage('__adgntseg');
```

Comment:
do you need consent from the user before reading from the local storage ?

Reply:
we're the only one adding to / reading from this variable, so it is only a cache for previously added segments - since the segments would be retrieved from the server here anyways, it's only an optimization